src/lib/FlightTasks/tasks/Orbit/FlightTaskOrbit.cpp

```diff
@@ -173,8 +180,39 @@ bool FlightTaskOrbit::update()
 
 	Vector2f center_to_position = Vector2f(_position) - _center;
 
-	// make vehicle front always point towards the center
-	_yaw_setpoint = atan2f(center_to_position(1), center_to_position(0)) + M_PI_F;
+	switch (_yaw_behavior) {
+	case 0:
+		// make vehicle front always point towards the center
+		_yaw_setpoint = atan2f(center_to_position(1), center_to_position(0)) + M_PI_F;
+		break;
+
+	case 1:
+		// make vehicle keep the same heading as in the beginning of the flight task
+		_yaw_setpoint = _initial_heading;
+		break;
+
+	case 2:
+		// no yaw setpoint
```

Comment:
Needs to be reset `_yaw_setpoint = NAN;` otherwise the last setpoint just stays.